autogpt_platform/backend/backend/server/routers/v1.py

```diff
@@ -592,35 +593,21 @@ def get_credentials(credentials_id: str) -> "Credentials | None":
 )
 async def execute_graph(
     graph_id: str,
-    node_input: Annotated[dict[str, Any], Body(..., default_factory=dict)],
     user_id: Annotated[str, Depends(get_user_id)],
+    inputs: Annotated[dict[str, Any], Body(..., embed=True, default_factory=dict)],
+    credentials_inputs: Annotated[
+        dict[str, CredentialsMetaInput], Body(..., embed=True, default_factory=dict)
+    ],
     graph_version: Optional[int] = None,
     preset_id: Optional[str] = None,
 ) -> ExecuteGraphResponse:
-    graph: graph_db.GraphModel | None = await graph_db.get_graph(
-        graph_id=graph_id, user_id=user_id, version=graph_version
-    )
-    if not graph:
-        raise ValueError(f"Graph #{graph_id} not found.")
-
-    graph_exec = await execution_db.create_graph_execution(
+    graph_exec = execution_utils.add_graph_execution(
         graph_id=graph_id,
-        graph_version=graph.version,
-        nodes_input=execution_utils.construct_node_execution_input(
-            graph, user_id, node_input
-        ),
         user_id=user_id,
+        inputs=inputs,
         preset_id=preset_id,
-    )
-
-    bus = execution_event_bus()
-    await bus.publish(graph_exec)
-
-    queue = await execution_queue_client()
-    await queue.publish_message(
-        routing_key=execution_utils.GRAPH_EXECUTION_ROUTING_KEY,
-        message=graph_exec.to_graph_execution_entry().model_dump_json(),
-        exchange=execution_utils.GRAPH_EXECUTION_EXCHANGE,
+        graph_version=graph_version,
+        graph_credentials_inputs=credentials_inputs,
     )
```

Comment:
Because I've replaced this boilerplate logic with `add_graph_execution(..)`, this is now a fully sync call. Any issues with that @majdyz?

Reply:
Discussed https://github.com/Significant-Gravitas/AutoGPT/pull/9789/files#r2048731698